pheget/api/__init__.py

```diff
@@ -2,13 +2,73 @@
 API endpoints (return JSON, not HTML)
 """
 
+import json
+import sqlite3
+import urllib.request
+
 from flask import Blueprint, abort, jsonify, request
 
+from .. import model
 from .format import query_variants
 
 api_blueprint = Blueprint("api", __name__)
 
 
+@api_blueprint.route("/bestvar/<string:symbol>/", methods=["GET"])
+def gene_query(symbol: str):
+    """
+    Given a gene, query an SQL database to find the variant with the strongest association with that gene
+    """
+    # ~UGLY HACK~ If omnisearch did not return a gene_id, then searchbox will fill in "undefined" - immediately abort in that case
+    if symbol == "undefined":
+        abort(404)
+    # Instead of querying locally, I will use omnisearch to get the relevant parameters (chrom, pos, gene_id)
+    url = f"https://portaldev.sph.umich.edu/api/v1/annotation/omnisearch/?q={symbol}&build=GRCh38"
+    omniJson = json.load(urllib.request.urlopen(url))["data"][0]
```

Comment:
If we *are* fetching data on the backend, remember to account for the possibility that the request would fail.

Reply:
My hope is that https://github.com/statgen/pheget/blob/265294173a518672df1a996c26a931892c73f202/pheget/api/__init__.py#L28-L33 will handle any failures.

Reply:
>  will handle any failures.

Writing distributed applications that depend on an external resource is hard. There are several kinds of failure to be concerned about. This isn't formal terminology, just examples I'm dashing off.

1. Hard failures: the other server is down
2. Soft failures: the other server is slow to respond for reasons outside your control, so this API takes a minute (may or may not ever complete)
3. Deterministic failures: eg the other server receives your request, but returns "400 bad request" in response to that input

For categories 1 & 2, I honestly forget what urllib does (I've generally used `requests`, which wraps some of the edge cases with nicer behaviors and can be told to raise exceptions etc for api errors). For category 3, checking the json payload might be sufficient.